hawkbit-ddi/hawkbit-ddi-api/src/main/java/org/eclipse/hawkbit/ddi/json/model/DdiActionFeedback.java

```diff
@@ -63,8 +66,10 @@ public class DdiActionFeedback {
     @JsonCreator
     public DdiActionFeedback(
             @JsonProperty(value = "time") final String time,
+            @JsonProperty(value = "timestamp") final Long timestamp,
```

Comment:
with jsoncreator/jackson we can hacve just one constructor-  timestamp is optional and can be skipped and initialized in constructor if null, i'll add however a DDI test that sends pure http request (not using Feign, just to be sure) - otherwise there is additional test to cover deserilization of json object without 'timestamp' field shouldConvertItWithoutOptionalFieldTimestamp

Reply:
yes - but the other constructor could be without @JsonCreator. To be used in tests and Feign

For instance DdiActionFeedback(Instant.now().toEpochMilli(), ddiStatus)) could be just
DdiActionFeedback(ddiStatus))

Also, since the status is mandatory - better be first argument here.

And finally - matter to unify timestamp by current millis - Instant.now().toEpochMilli()  or System.currentTimeMillis() - check what is used elsewhere in hawkbit. If there is preferable approach - better use it at all places

Reply:
got it